components/internal/modals/EditPaymentDetailsModal.tsx

```diff
@@ -0,0 +1,357 @@
+  const [shippingCity, setShippingCity] = useState('');
+  const [shippingProvince, setShippingProvince] = useState('');
+  const [shippingCountry, setShippingCountry] = useState('');
+  const [shippingPostalCode, setShippingPostalCode] = useState('');
+
+  //   TODO: Add error states for each field as follows
+  // const [shippingFullNameInputError, setShippingFullNameInputError] = useState(''); // Error message displayed under input
+
+  //   Billing address information state
+  const [hideBillingInfo, setHideBillingInfo] = useState(false); // Whether billing information is visible
+  const [billingFullName, setBillingFullName] = useState('');
+  const [billingAddressLine1, setBillingAddressLine1] = useState('');
+  const [billingAddressLine2, setBillingAddressLine2] = useState('');
+  const [billingCity, setBillingCity] = useState('');
+  const [billingProvince, setBillingProvince] = useState('');
+  const [billingCountry, setBillingCountry] = useState('');
+  const [billingPostalCode, setBillingPostalCode] = useState('');
+
+  const handleSubmit = (event: SyntheticEvent) => {
+    event.preventDefault();
+    // TODO: Add error handling to each field as follows
+    // if (!hideShippingInfo) {
+    //   if (!shippingFullName.length) {
+    //     setShippingFullNameInputError("Please enter the recipient's full name.");
+    //   }
+    // }
+  };
+
+  return (
+    <>
+      {/* Button will be removed before merging */}
+      <Button mt={3} onClick={onOpen}>
+        Open
+      </Button>
+      <Modal
+        onClose={onClose}
+        isOpen={isOpen}
+        scrollBehavior="inside"
+        size="3xl" // TODO: change to custom size
+      >
+        <ModalOverlay />
+        <form onSubmit={handleSubmit}>
+          <ModalContent paddingX="40px">
+            <ModalHeader textStyle="display-medium-bold" paddingBottom="12px" paddingTop="24px">
+              <Text textStyle="display-medium-bold">
+                {'Edit Payment, Shipping and Billing Details'}
+              </Text>
+            </ModalHeader>
+            <ModalBody paddingY="20px">
+              <Box paddingBottom="32px">
+                <Grid templateColumns="repeat(2, 1fr)" rowGap={'24px'}>
+                  <GridItem rowSpan={2} colSpan={1}>
+                    <FormControl as="fieldset" isRequired>
+                      <FormLabel as="legend">{'Payment method'}</FormLabel>
+                      <RadioGroup onChange={setPaymentMethod} value={paymentMethod}>
+                        <Stack>
+                          <Radio value="Mastercard">{'Mastercard'}</Radio>
+                          <Radio value="Visa">{'Visa'}</Radio>
+                          <Radio value="Debit">{'Debit'}</Radio>
+                          <Radio value="Cash">{'Cash'}</Radio>
+                          <Radio value="Cheque">{'Cheque'}</Radio>
+                          <Radio value="E-transfer">{'E-transfer'}</Radio>
```

Comment:
Addressed! I had to make it `const [paymentMethod, setPaymentMethod] = useState<PaymentType | string>('');` instead because onChange (code snippet below) cannot accept null/undefined values and was throwing errors. Lmk if there are any issues with that


``` ts
                      <RadioGroup onChange={setPaymentMethod} value={paymentMethod}>
```

Reply:
Oh, okay. Does doing `useState<PaymentType | ''>('');` work?

Reply:
![image](https://user-images.githubusercontent.com/51224641/126722909-5fa05d02-3923-4073-80f1-d4e52c278028.png)

😬 throws an error

Reply:
Okay, let's leave it as what it is right now then!